OpenEdXMobile/src/main/java/org/edx/mobile/module/registration/view/RegistrationOptionSpinner.java

```diff
@@ -23,23 +28,51 @@ public RegistrationOptionSpinner(Context context, AttributeSet attrs, int defSty
         super(context, attrs, defStyle);
     }
 
-    public boolean hasValue(String value){
-        if ( value == null || items == null )
-            return false;
-        for ( RegistrationOption option : items ) {
-            if ( value.equals(option.getValue()) )
-                return true;
+    public boolean hasValue(@Nullable String value){
+        boolean ret = false;
+        if (adapter != null) {
+            ret = (getAdapterPosition(value) >= 0);
+        }
+        return ret;
+    }
+
+    public void select(@Nullable String value){
+        if (adapter != null && value != null) {
+            int pos = getAdapterPosition(value);
+            if (pos >= 0) {
+                setSelection(pos);
+            }
+        }
+    }
+
+    public @Nullable String getSelectedItemValue() {
+        String ret = null;
+        RegistrationOption selected = (RegistrationOption)getSelectedItem();
+        if (selected != null) {
+            ret = selected.getValue();
         }
-        return false;
+        return ret;
     }
 
-    public void select(String value){
-        if ( value == null || items == null )
-            return;
-        for ( RegistrationOption option : items ) {
-            if ( value.equals(option.getValue()) ){
-                super.select(option);
+    private int getAdapterPosition(@Nullable String input) {
+        int ret = -1;
+        if (input != null && adapter != null) {
+            for(int i=0 ; i<adapter.getCount() ; i++){
+                RegistrationOption item = adapter.getItem(i);
+                if (item != null && input.equals(item.toString())) {
+                    ret = i;
+                    break;
+                }
             }
         }
+        return ret;
```

Comment:
It's the name of the variable that will be returned. Again, I will change it, but is this really necessary to hold up a PR?